tooling/noirc_abi/src/input_parser/json.rs

```diff
@@ -121,14 +121,13 @@ impl JsonTypes {
     /// A hex string's length needs to be even to decode into bytes, as two digits correspond to
     /// one byte.
     fn format_field_string(field: FieldElement) -> String {
-        if field.is_zero() {
-            return "0x00".to_owned();
-        }
-        let mut trimmed_field = field.to_hex().trim_start_matches('0').to_owned();
-        if trimmed_field.len() % 2 != 0 {
-            trimmed_field = "0".to_owned() + &trimmed_field;
-        }
-        "0x".to_owned() + &trimmed_field
+        let trimmed = field.to_hex().trim_start_matches('0').to_owned(); // empty when field is 0
+
+        // select "00" when field is 0, "" when trimmed field length is even, and "0" when the length is odd
+        let chars = ["00", "", "0"];
+        let index = (!trimmed.is_empty() as usize) + (trimmed.len() & 1);
+
+        format!("0x{}{}", chars[index], trimmed)
```

Comment:
Was there a problem with the previous code? The new code seems more clever but less readable so I prefer the old snippet.